ngraph/src/ngraph/frontend/onnx_import/op/round.cpp

```diff
@@ -30,10 +30,44 @@ namespace ngraph
         {
             namespace set_1
             {
+                // WARNING!
+                // Current version is:
+                // data_floor = floor(data)
+                // diff = data - data_floor
+                // if(diff < 0.5f)
+                //   return data_floor
+                // else
+                //   return data_floor + 1.0f
+                //
+                // The correct version should contain condition:
+                // if (diff < 0.5f || (diff == 0.5f && static_cast<int>(data_floor) % 2 == 0))
                 OutputVector round(const Node& node)
                 {
                     const Output<ngraph::Node> data{node.get_ng_inputs().at(0)};
-                    return {std::make_shared<ngraph::opset0::Round>(data)};
+
+                    Output<ngraph::Node> one_const =
+                        default_opset::Constant::create(element::f32, {}, {1.0f});
+                    Output<ngraph::Node> half_const =
+                        default_opset::Constant::create(element::f32, {}, {0.5f});
+                    if (data.get_element_type() != element::f32)
+                    {
+                        one_const = std::make_shared<default_opset::Convert>(
+                            one_const, data.get_element_type());
+                        half_const = std::make_shared<default_opset::Convert>(
+                            half_const, data.get_element_type());
+                    }
+
+                    const auto data_floor = std::make_shared<default_opset::Floor>(data);
+                    const auto data_floor_plus_one =
+                        std::make_shared<default_opset::Add>(data_floor, one_const);
+
+                    const auto diff = std::make_shared<default_opset::Add>(
+                        data, std::make_shared<default_opset::Negative>(data_floor));
```

Comment:
Why are you adding the negative `data_floor` instead of simple subtraction?

Reply:
right ;)